src/backend/executor/nodeModifyTable.c

```diff
@@ -3654,6 +3709,70 @@ ExecEndModifyTable(ModifyTableState *node)
 	 * shut down subplan
 	 */
 	ExecEndNode(outerPlanState(node));
+
+	/* Notify modified leaf relids to QD */
+	if (GP_ROLE_EXECUTE == Gp_role && node->has_leaf_changed)
+		notify_modified_relations_to_QD(node);
```

Comment:
What's the behavior of modified status propagation in single-node or utility mode?